src/top/data/bbox_reg_util.py

```diff
@@ -86,72 +89,85 @@ class CropObject(object):
+
+        scale = th.as_tensor(scale)
+        scale = scale.reshape(-1,3)
 
-        h, w = image.shape[-2:]
-        keypoints_2d = np.split(inputs[Schema.KEYPOINT_2D], np.array(np.cumsum(num_keypoints)))
-        keypoints_2d = [points.reshape(-1,3) for points in keypoints_2d]
-        keypoints_2d = [np.multiply(keypoint, np.array([w, h, 1.0], np.float32)).astype(int)
-                        for keypoint in keypoints_2d]
+        translation = th.as_tensor(translation)
+        translation = translation.reshape(-1,3)
+
+        point_min = th.min(keypoints_2d_clamp, dim=1).values
+        point_max = th.max(keypoints_2d_clamp, dim=1).values
+        
+        vis_img = th.tensor([])
+        vis_crop_img = th.tensor([])
+        vis_point_2d = th.tensor([])
+        vis_trans = th.tensor([])
+        vis_orient = th.tensor([])
+        vis_scale = th.tensor([])
         
-        orientation = np.split(orientation, num_object)
-        orientation = [rotation.reshape(3,3) for rotation in orientation]
-
-        scale = np.split(scale, num_object)
-        scale = [scales.reshape(-1,3) for scales in scale]
-
-        translation = np.split(translation, num_object)
-        translation = [translations.reshape(-1,3) for translations in translation]
-
-        crop_img = []
-        quaternions = []
-        _scale = []
-        _translation = []
-        for object_id in range(num_object):       
-            # NOTE(Jiyong): np.split() leaves an empty array at the end of the list.
-            if keypoints_2d[object_id].size == 0:
-                break
-
-            x_min, y_min, _ = np.min(keypoints_2d[object_id], axis=0)
-            x_max, y_max, _ = np.max(keypoints_2d[object_id], axis=0)
-
-            # NOTE(Jiyong): TypeError: Expected cv::UMat for argument 'src'
-            # -> cv2.Umat() is functionally equivalent to np.float32() & (H,W,C)
-            crop_tmp = np.float32(image[:, y_min:y_max, x_min:x_max])
-            crop_tmp = np.transpose(crop_tmp, (1,2,0))
-            try:
-                crop_tmp = cv2.resize(crop_tmp, dsize=self.opts.crop_img_size)
-            except Exception as e:
-                print(crop_tmp.shape)
-                raise
-
-            crop_tmp = np.transpose(crop_tmp, (2,0,1))
-            crop_img.append(crop_tmp)
-
-            # For quaternions regression
-            r = R.from_matrix(orientation[object_id])
-            quaternions.append(r.as_quat())
-
-            _scale.append(scale[object_id])
-            _translation.append(translation[object_id])
+        for obj in range(num_object):
+            # NOTE(Jiyion): If visiblity is false, cropping that object is impossible
+            if not visibility[obj]:
+                continue
 
+            crop_tmp = image[:, int(point_min[obj][1]):int(point_max[obj][1]), int(point_min[obj][0]):int(point_max[obj][0])]
+            crop_tmp = resize(crop_tmp, size=self.opts.crop_img_size)
+
+            vis_img = th.cat((vis_img, image))
```

Comment:
What's the point of `vis` prefix? All these tensors are surely not just for visualization?
Also, it would be preferable not to incrementally build out a tensor which would cause repeated allocation/de-allocation of memory